### ISSUE TITLE
spherical_jn: silently wrong results in float32

### ISSUE THREAD

**Issue by nstarman** [JAXtronomy/spexial#51]
Follow-up to #50.

`spherical_jn` / `spherical_jn_all` use upward recurrence with a turning-point cutoff (`_CUTOFF = 2e-10`) tuned for float64. Under float32 the results are silently wrong: finite numbers, no nan, no warning.

Measured against `scipy.special.spherical_jn` on `z ∈ [0.01, 2n]` in float32:

| n | max abs error / max\|j_n\| |
|---|---|
| 5 | 8.1 |
| 20 | 80 |
| 100 | 51 |

The PR body notes that float64 is required, but neither the docstrings nor `docs/reference/accuracy-and-domains.md` say so.

Options:
- compute internally in float64 and cast back, as `comb`/`k0` already do for float16/bfloat16 (this needs x64 enabled);
- or raise an error for non-float64 input, and document it;
- or derive a dtype-dependent cutoff from `finfo(dtype).eps`. This still leaves float32 accuracy poor near the turning point.

Repro:
```python
import jax.numpy as jnp, numpy as np, scipy.special as ss, spexial as sp
x = np.linspace(0.01, 40, 2001).astype(np.float32)
a = np.asarray(sp.spherical_jn(20, jnp.asarray(x)))
b = ss.spherical_jn(20, x.astype(float))
print(np.max(np.abs(a - b)) / np.max(np.abs(b)))  # ~80
```
